### PR TITLE
Push all datapackets

### PR DESCRIPTION
**Pull request by lasdasdas**
Hi, we added yet another macro. If you consider this is not useful to your project, its not a problem, we can fork the project and change it to fit our needs. I'll still send you PR meanwhile as these changes might be useful to someone else. 

The macro forces the drivers to push the packets to the queue as soon as possible and not to wait for the whole frame. 

Cheers!

### CODE REVIEW

File: sample/simple/CMakeLists.txt [hunk 1]
```diff
@@ -63,6 +63,13 @@ if( USE_MILLIMETERS )
   add_definitions( -DUSE_MILLIMETERS )
 endif()
 
+# Push data packets as soon as possible, if ndef the drivers push the 
+# whole frame
+OPTION( PUSH_SINGLE_PACKETS  "Push data packets to the stack instead of frames" OFF )
```
Comment (UnaNancyOwen) [inline sample/simple/CMakeLists.txt:68]: Please turn on the default option. (It keeps the same operation as before by default.)

**Review by UnaNancyOwen — COMMENTED** [review]
Thank you for your contributions.
As a basic policy, Please keeps the same operation as before by default.

File: sample/simple/CMakeLists.txt (hunk 1 above)
Comment (lasdasdas) [reply sample/simple/CMakeLists.txt:68]: PUSH_SINGLE_PACKETS is OFF therefore 
```
                            #ifndef PUSH_SINGLE_PACKETS
                            if( last_azimuth > azimuth ){
                            if( last_azimuth > azimuth ){
                                // Push One Rotation Data to Queue
                                // Push One Rotation Data to Queue
                                mutex.lock();
                                mutex.lock();
                                queue.push( std::move( lasers ) );
                                queue.push( std::move( lasers ) );
                                mutex.unlock();
                                mutex.unlock();
                                lasers.clear();
                                lasers.clear();
                            }
                            }
                            #endif
```
Is activated by default and  It's the same behavior as before right?

Comment (UnaNancyOwen) [reply sample/simple/CMakeLists.txt:68]: Oh! Sorry, I misunderstood. You're right. 
I understand about this feature now.

**Comment by UnaNancyOwen** [comment]
LGTM 👍

**Comment by UnaNancyOwen** [comment]
Thank you for your contribution! I think this feature is good and very useful for many users.

**Comment by lasdasdas** [comment]
Thank you @UnaNancyOwen  :smile:

**Pull request merged**